primer-io/primer-sdk-react-native
# fix: Fixes incorrect type mapping in onBeforePaymentCreate callback on iOS. BREAKING CHANGE

[DPS-300](https://primerapi.atlassian.net/browse/DPS-300)

This PR fixes the incorrect type being sent through the ReactNative interface for `onBeforePaymentCreate`

This is a breaking change 

We map the data with the same strategy seen on Android: [here](https://github.com/primer-io/primer-sdk-react-native/blob/master/packages/sdk/android/src/main/java/com/primerioreactnative/extensions/PrimerPaymentMethodData.kt) and [here](https://github.com/primer-io/primer-sdk-react-native/blob/master/packages/sdk/android/src/main/java/com/primerioreactnative/datamodels/PrimerPaymentMethodDataRN.kt))

To test:
Set a breakpoint in `primerWillCreatePaymentWithData` in `RNTPrimerHeadlessUniversalCheckout.swift` and `RNTPrimer.swift` and observe the type being passed back.

[DPS-300]: https://primerapi.atlassian.net/browse/DPS-300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
